### DISCUSSION TITLE
I'm confused,i wanna upload some info to server,about 2 images and some details info. error:"msg": "str type expected",       "type": "type_error.str"

### DISCUSSION THREAD

**Discussion by Mecil9** [fastapi/fastapi#8417] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
...
class ResponseSchema(BaseModel):
    """Response Model"""
    detail: str
    result: Optional[T] = None

class BannerInputSchema(BaseModel):
    """Banner Create Model"""
    title: str
    sub_title: str | None = None
    path_url: str | None = None
    desc: str | None = None

    class Config:
        orm_mode = True

@app.put('/update_banner/{id}',
            response_model=ResponseSchema,
            response_model_exclude_none=True)
async def update_banner(
        banner_id: str = Form(...),
        banner_info: BannerInputSchema = Form(...),
        b_img: UploadFile = File(default=None, description='upload background img'),
        f_img: UploadFile = File(default=None, description='upload front img'),
):
    result = await BannerService.update_banner(
        banner_id=banner_id,
        banner_info=banner_info,
        b_img=b_img,
        f_img=f_img,
    )

    return ResponseSchema(detail='Success Banner Data', result=result)
```

### Description

I am tryting to upload 2 images and some detail info of Banner(ex:id,title,subtitle,desc...). I used Form,UploadFile,When i use Schema model ,it throw an error. if i single input per attribute,it's work,how to fix it? Thanks.

{
  "detail": [
    {
      "loc": [
        "body",
        "banner_info",
        "title"
      ],
      "msg": "str type expected",
      "type": "type_error.str"
    }
  ]
}
if i use Body():
...
banner_id: str = Body(embed=True),
        banner_info: BannerInputSchema = Body(embed=True),
        b_img: UploadFile = File(default=None, description='upload background img'),
        f_img: UploadFile = File(default=None, description='upload front img'),
...
it's more error about upload image.
{
  "detail": [
    {
      "loc": [
        "body",
        "banner_info",
        "title"
      ],
      "msg": "str type expected",
      "type": "type_error.str"
    },
    {
      "loc": [
        "body",
        "b_img"
      ],
      "msg": "Expected UploadFile, received: <class 'str'>",
      "type": "value_error"
    },
    {
      "loc": [
        "body",
        "f_img"
      ],
      "msg": "Expected UploadFile, received: <class 'str'>",
      "type": "value_error"
    }
  ]
}

### Operating System

macOS

### Operating System Details

macOs m1 13.0.1

### FastAPI Version

fastapi = "^0.86.0"

### Python Version

python.3.9.12

**Comment by mateoradman** [comment, marked as answer]
The problem is that you can't use Pydantic models for validation of Form fields. 
You can solve it like this
```python
class AnyForm(BaseModel):
    any_param: str
    any_other_param: int = 1

    @classmethod
    def as_form(
        cls,
        any_param: str = Form(...),
        any_other_param: int = Form(1)
    ) -> AnyForm:
        return cls(any_param=any_param, any_other_param=any_other_param)

@router.post('')
async def any_view(form_data: AnyForm = Depends(AnyForm.as_form)):
        ...
```

Source: https://stackoverflow.com/a/60670614

**Comment by dotX12** [comment]
@Mecil9, @mateoradman https://github.com/dotX12/pyfa-converter a little smarter, prettier and more productive :)
